src/LinAlg/hiopLinAlgFactory.cpp

```diff
@@ -204,22 +204,36 @@ hiopMatrixSparse* LinearAlgebraFactory::createMatrixSymSparse(int size, int nnz)
 /**
  * @brief Static method to create a raw C array
  */
-double* LinearAlgebraFactory::createRawArray(int n)
```

Comment:
I decided to add a default value parameter to this method in order to allow for a default value for the memory to be initialised too. I support this as a change as it prevents potential unwanted behaviour resulting from compiler default initialisation values. This being said, this does result in an extra memory transfer occurring in order to incorporate the functionality.

Since this functionality is only being added in order to better accommodate testing, I can see how it may be unnecessary as it is worse for performance. Without this change, I do not see a way to remove the createLocalBuffer and deleteLocalBuffer method implementations for the vector testing suite, so I would need to re-impliment the changes.

I also would have used an optional function parameter if we were able to use c++17 an onwards...

Reply:
> this does result in an extra memory transfer occurring in order to incorporate the functionality.

I don't think it does from what I can see. I would only add a default value for `value` like:
```cpp
double* LinearAlgebraFactory::createRawArray(int n, int value=0)
```
so we don't have to change existing code.

Reply:
There are two buffers that are created in the modified implementation [here](https://github.com/LLNL/hiop/blob/459d5493bbfc3b4be0bfbdf1512adc7e3141a2ed/src/LinAlg/hiopLinAlgFactory.cpp#L225), along with a call to `resmgr.copy()`. In the previous implementation, only one allocation is made, and no copies are performed.

Reply:
Since this PR is for unit tests refactoring, I think these changes are out of scope here. There should be a separate issue/PR if you think we need to make these changes to the source code. Same applies for the rest of the changes in `src`.

Reply:
Moved changes in `src` to #157